common_defs.h

```diff
@@ -67,6 +67,28 @@ typedef int64_t s64;
 #  endif
 #endif
 
+#if defined(_MSC_VER)
+#  if defined(_M_X64)
+#    define ARCH_X64
+#  elif defined(_M_IX86)
+#    define ARCH_X86
+#  elif defined(_M_ARM64)
+#    define ARCH_ARM64
+#  elif defined(_M_ARM)
+#    define ARCH_ARM
```

Comment:
The compiler arch macros (both the MSVC ones and the gcc/clang ones) are notoriously unclear about what is 32-bit and what is 64-bit.  As long as this abstraction layer is being introduced, how about using super clear names:

* ARCH_ARM32
* ARCH_ARM64
* ARCH_X86_32
* ARCH_X86_64

Reply:
addressed